src/freenet/client/filter/TheoraPacketFilter.java

```diff
@@ -249,6 +206,15 @@ private void verifyTypeAndHeader(BitInputStream input, int expectedHeaderType) t
         }
     }
 
+    private int checkHeaderField(String headerName, String fieldName,
+                                 BitInputStream input, int sizeInBits, Predicate<Integer> validator) throws IOException {
+        int value = input.readInt(sizeInBits);
+        if (validator.test(value)) {
```

Comment:
A validator should return `true`if the given value is valid, something needs to be the other way around here. :)